capa/main.py

```diff
@@ -780,6 +807,28 @@ def get_extractor_from_cli(args, input_format: str, backend: str) -> FeatureExtr
         raise ShouldExitError(E_CORRUPT_FILE) from e
 
 
+def get_extractor_filters_from_cli(args, input_format) -> Dict[str, Set]:
+    if input_format in STATIC_FORMATS:
+        if args.restrict_to_processes:
+            raise InvalidArgument("Cannot filter processes with static analysis.")
+        return {"functions": {int(addr, 0) for addr in args.restrict_to_functions}}
+    elif input_format in DYNAMIC_FORMATS:
+        if args.restrict_to_functions:
+            raise InvalidArgument("Cannot filter functions with dynamic analysis.")
+        return {"processes": {int(pid, 0) for pid in args.restrict_to_processes}}
+    else:
+        raise ShouldExitError(E_INVALID_INPUT_FORMAT)
+
+
+def apply_extractor_filters(extractor: FeatureExtractor, extractor_filters: Dict[str, Set]):
+    if isinstance(extractor, StaticFeatureExtractor):
+        return FunctionFilter(extractor, extractor_filters["functions"])
```

Comment:
should check to see that `functions` is present and non-empty. ditto with `processes`.

Reply:
because i'm thinking about the future where we may add additional filters, so that its not implicit that "if is static and there are filters then it must be a function filter".

Reply:
> because i'm thinking about the future where we may add additional filters, so that its not implicit that "if is static and there are filters then it must be a function filter".

I've added this now. I am doing it with an assert now however, since I imagine that if we add a basic blocks filter for example then we would have an if-elif-else statement there or somewhere else to check the fields.